.github/workflows/appmap-analysis.yml

```diff
@@ -0,0 +1,57 @@
+name: appmap-analysis
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main # Change this to the name of the mainline branch
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      # Install AppMap library and tools
+      - name: Enable appmap gem update
+        run: bundle config unset deployment      
+      - name: Install AppMap tools
+        uses: getappmap/install-action@v1
+        with:
+          project-type: bundler # Choose the type of your project here:
+                        # bundler, maven, gradle, pip, pipenv, poetry,
+                        # yarn, npm, etc.
+ 
+      ####################################################################
+      # Put the commands required to run your application test suite.
+      # For example, create and setup the test database, commands 
+      # to run your test suite.
+      - name: Rails test setup
+        run: ./bin/rails db:migrate
+
+      - name: Run tests
+        run: ./bin/rails test
+      
+      - name: Save AppMaps
+        uses: actions/cache/save@v3
+        if: always()
+        with:
+          path: tmp/appmap
+          key: appmaps-${{ github.sha }}-${{ github.run_attempt }}
+
+  appmap-analysis:
+    if: always()
+    needs: [test] # You may need to change this to match the name of the step that runs your tests.
+    uses: getappmap/analyze-action/.github/workflows/appmap-analysis.yml@feat/workflow-run-file
```

Comment:
Should work as-is with the two changes above. No appmap-dir input is needed to the reusable workflow at this time.